source/hello-world.html.haml.markdown

```diff
@@ -11,98 +11,164 @@ description: Create a Flatpak application in five minutes.
-  And then install the runtime itself:
+  In addition to the runtime itself, there is a related Sdk (Software Development Kit), which contains all the things that
+  are in the runtime, and additionally all that you need to build applications against it. This includes compilers, development
+  tools, headers and similar things that are typically packages in -devel/-dev packages on traditional distributions.
 
   <pre>
-  <span class="unselectable">$ </span>flatpak install gnome org.gnome.Platform//3.22
+  <span class="unselectable">$ </span>flatpak install flathub org.freedesktop.Platform//1.6 org.freedesktop.Sdk//1.6
   </pre>
-          
+
   ## 3. Create your app
 
-  First we need to create the directory structure for the app:
+  In this example we're doing something very simple, so add the following to a file and save it as `hello.sh`:
 
   <pre>
-  <span class="unselectable">$ </span>mkdir hello
-  <span class="unselectable">$ </span>mkdir hello/files
-  <span class="unselectable">$ </span>mkdir hello/files/bin
-  <span class="unselectable">$ </span>mkdir hello/export
+  #!/bin/sh
+  echo "Hello world, from a sandbox"
   </pre>
 
-  Now let's add something simple: create a file in `hello/files/bin/` called `hello.sh`. Then add the following to the file and save it:
+  In a more typical scenario you already have an application that you want to package.
+
+  ## 4. Name your application
+
+  In flatpak applications are refered to by their application id, which is a unique identifier in reverse-DNS style.
+  For example `org.gnome.gedit` or `net.supertuxkart.SuperTuxKart`. Chosing the right id is important, as it is hard to
+  change this later. The rules for flatpak names are the same as those for dbus names, and if your application is using
+  dbus then the id should should match the dbus name.
+
+  If the application doesn't have a dbus name you should pick a name based on some dns namespace you own. Projects often already
+  have a web presence, or are related to some project that does. If the project is on github you can use the user.github.io
+  domain that gives you.
+
+  In this example we will use the name `org.flatpak.Hello`.
+
+  ## 5. Write a manifest
+
+  Most flatpak applications are build using a tool called flatpak-builder. flatpak-builder takes a description of the build
+  called a manifest, which is a json file. So, add the following to a file called `org.flatpak.Hello.json`:
 
   <pre>
-  #!/bin/sh
-  echo "Hello world, from a sandbox"
+  {
+      "app-id": "org.flatpak.Hello",
+      "runtime": "org.freedesktop.Platform",
+      "runtime-version": "1.6",
+      "sdk": "org.freedesktop.Sdk",
+      "command": "hello.sh",
+      "modules": [
+          {
+              "name": "hello",
+              "buildsystem": "simple",
+              "sources": [
+                  {
+                      "type": "file",
+                      "path": "hello.sh"
+                  }
+              ],
+              "build-commands": [
```

Comment:
Minor detail but convention seems to be build-commands and configuration going before sources since sources tends to be the longest section.